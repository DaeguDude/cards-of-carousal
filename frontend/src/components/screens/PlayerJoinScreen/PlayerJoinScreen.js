import React, { useState } from 'react';
import styled from 'styled-components';
import Button from '../../Buttons/Button';
import Header from '../../Header/Header';
import Footer from '../../Footer/Footer';

const propTypes = {};

const PlayerJoinContainer = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: grid;
  height: 100%;
  background-color: var(--primary-background-color);
  grid-template-rows: 190px 1fr 140px;
  grid-template-columns: 1fr;

  main {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .player-join-form-container {
    width: 100%;
    max-width: 1650px;
    margin: 0 auto;
    text-align: center;
    padding: 20px 80px 20px 80px;
  }

  .player-join-form {
    display: grid;
    grid-template-columns: 52% 48%;
    grid-template-rows: 90px 160px 1fr;
  }

  input {
    text-transform: uppercase;
    font-size: 4.4rem;
    font-weight: 500;

    border: none;
    border-bottom: 4px solid var(--primary-text-color);
    padding-top: 24px;
    background: transparent;
    padding-left: 0;
  }

  input::placeholder {
    color: var(--primary-text-color);
    font-weight: 500;
    font-size: 4.4rem;
  }

  .player-join-name-input {
    margin-right: 32px;
  }

  .player-join-code-input {
    margin-left: 32px;
  }

  // standard screen sizes
  @media screen and (max-width: 2000px) {
    .player-join-form-container {
      max-width: 1000px;
    }

    .player-join-form {
      grid-template-rows: 60px 88px 1fr;
    }

    input {
      font-size: 1.6rem;
      border-bottom: 2px solid var(--primary-text-color);
      padding-top: 36px;
    }

    input::placeholder {
      font-size: 1.6rem;
    }

    .player-join-name-input {
      margin-right: 16px;
    }

    .player-join-code-input {
      margin-left: 16px;
    }
  }

  @media screen and (max-width: 1024px) {
    .player-join-form-container {
      max-width: 800px;
    }

    .player-join-form {
      grid-template-rows: 60px 80px 1fr;
    }
  }

  @media screen and (max-width: 970px) {
    .player-join-form {
      grid-template-rows: 60px 60px 1fr;
    }

    input {
      font-size: 1.2rem;
      border-bottom: 2px solid var(--primary-text-color);
      padding-top: 40px;
    }

    input::placeholder {
      font-size: 1.2rem;
    }
  }

  @media screen and (max-width: 785px) {
    // header, main, footer
    grid-template-rows: 180px 1fr 120px;

    .player-join-form-container {
      margin-top: -88px;
    }
  }

  @media screen and (max-width: 646px) {
    .player-join-form-container {
      padding-left: 40px;
      padding-right: 40px;
    }

    .player-join-form {
      grid-template-columns: 1fr;
      grid-template-rows: 60px 60px 40px 1fr;
    }

    .player-join-name-input {
      grid-row: 1;
      grid-column: 1;
    }

    .player-join-code-input {
      grid-row: 2;
      grid-column: 1;
    }

    .player-join-name-input,
    .player-join-code-input {
      margin: 0 16px;
    }
  }

  @media screen and (max-width: 375px) {
    .player-join-form-container {
      padding: 20px;
    }
  }

  @media screen and (max-width: 365px) {
    // header, main, footer
    grid-template-rows: 130px 1fr 140px;
  }
`;

const PlayerJoinHeader = styled(Header)`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  // CoC logo
  h1 {
    font-size: 6rem;
    line-height: 6rem;
    text-transform: uppercase;

    margin-bottom: -16px;
    vertical-align: bottom;
  }

  @media screen and (max-width: 970px) {
    h1 {
      font-size: 4rem;
      line-height: 5rem;

      margin-bottom: -20px;
    }
  }

  @media screen and (max-width: 785px) {
    h1 {
      font-size: 4rem;
      line-height: 4.5rem;

      margin-bottom: -15px;
    }
  }

  @media screen and (max-width: 365px) {
    h1 {
      font-size: 3rem;
      line-height: 3.5rem;

      margin-bottom: -12px;
    }
  }
`;

const PlayerJoinButton = styled(Button)`
  grid-column: 1 / 3;
  grid-row: 3;

  width: 900px;
  height: 250px;
  margin: 0 auto;

  border-radius: 2px;
  background-color: var(--secondary-background-color);

  text-transform: uppercase;
  font-size: 4rem;

  @media screen and (max-width: 2000px) {
    font-size: 1.5rem;
    width: 360px;
    height: 100px;
  }

  @media screen and (max-width: 970px) {
    font-size: 1.2rem;
    width: 288px;
    height: 80px;
  }

  @media screen and (max-width: 785px) {
    grid-row: 4;
  }

  @media screen and (max-width: 450px) {
    width: 260px;
    height: 72px;
  }
`;

export default function PlayerJoinScreen() {
  const [name, setName] = useState('');
  const [joinCode, setJoinCode] = useState('');

  function handleSubmit(e) {
    e.preventDefault();
    // TODO - handle form submission by taking user to waiting screen
  }

  return (
    <PlayerJoinContainer
      data-testid="player-join-container"
      className="primary-background"
    >
      <PlayerJoinHeader>
        <h1>Cards of Carousal</h1>
      </PlayerJoinHeader>
      <main>
        <div className="player-join-form-container">
          <form onSubmit={(e) => handleSubmit(e)} className="player-join-form">
            <input
              type="text"
              placeholder="name"
              value={name}
              onChange={(e) => setName(e.target.value)}
              className="player-join-name-input"
            />
            <input
              type="text"
              placeholder="join code"
              value={joinCode}
              onChange={(e) => setJoinCode(e.target.value)}
              className="player-join-code-input"
            />
            <PlayerJoinButton type="submit">
              Let&apos;s Carouse!
            </PlayerJoinButton>
          </form>
        </div>
      </main>
      <Footer />
    </PlayerJoinContainer>
  );
}

PlayerJoinScreen.propTypes = propTypes;
