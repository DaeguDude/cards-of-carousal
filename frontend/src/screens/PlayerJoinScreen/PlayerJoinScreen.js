import React, { useContext, useState } from 'react';
import styled from 'styled-components';
import Button from '../../components/Buttons/Button';
import Header from '../../components/Header/Header';
import Footer from '../../components/Footer/Footer';
import { PlayerContext } from '../../contexts/PlayerContext/PlayerContext';

const propTypes = {};

const PlayerJoinContainer = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: grid;
  height: 100%;
  background-color: var(--primary-background-color);
  grid-template-rows: 130px 1fr auto;
  grid-template-columns: 1fr;

  main {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .player-join-form-container {
    width: 100%;
    max-width: 1650px;
    margin: 0 auto;
    text-align: center;
    padding: 20px;
    margin-top: -88px;
  }

  .player-join-form {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 60px 60px 40px 1fr;
  }

  .player-join-name-input {
    grid-row: 1;
    grid-column: 1;
  }

  .player-join-code-input {
    grid-row: 2;
    grid-column: 1;
  }

  .player-join-name-input,
  .player-join-code-input {
    margin: 0 16px;
  }

  input {
    text-transform: uppercase;
    font-size: 1.2rem;
    border: none;
    border-bottom: 2px solid var(--primary-text-color);
    padding-top: 40px;
    background: transparent;
    padding-left: 0;
  }

  input::placeholder {
    color: var(--primary-text-color);
    font-size: 1.2rem;
  }

  @media screen and (min-width: 646px) {
    .player-join-form-container {
      padding-left: 40px;
      padding-right: 40px;
    }

    .player-join-form {
      grid-template-columns: 52% 48%;
      grid-template-rows: 60px 60px 40px 1fr;
    }

    .player-join-name-input {
      grid-row: 1;
      grid-column: 1 / 3;
    }

    .player-join-code-input {
      grid-row: 2;
      grid-column: 1 / 3;
    }

    .player-join-name-input,
    .player-join-code-input {
      margin: 0 16px;
    }
  }

  @media screen and (min-width: 786px) {
    // header, main, footer
    grid-template-rows: 180px 1fr auto;

    .player-join-form-container {
      max-width: 800px;
    }

    .player-join-form {
      grid-template-rows: 60px 60px 1fr;
    }

    .player-join-name-input {
      grid-row: 1;
      grid-column: 1;
    }

    .player-join-code-input {
      grid-row: 1;
      grid-column: 2;
    }
  }

  @media screen and (min-width: 970px) {
    input {
      font-size: 1.2rem;
      border-bottom: 2px solid var(--primary-text-color);
      padding-top: 40px;
    }

    input::placeholder {
      font-size: 1.2rem;
    }
  }

  @media screen and (min-width: 2000px) {
    .player-join-form-container {
      max-width: 1000px;
    }

    .player-join-form {
      grid-template-rows: 60px 88px 1fr;
    }

    input {
      font-size: 1.6rem;
      border-bottom: 2px solid var(--primary-text-color);
      padding-top: 36px;
    }

    input::placeholder {
      font-size: 1.6rem;
    }

    .player-join-name-input {
      margin-right: 16px;
    }

    .player-join-code-input {
      margin-left: 16px;
    }
  }

  // 4k
  @media screen and (min-width: 2560px) {
    .player-join-form-container {
      max-width: 1650px;
      padding: 20px 80px 20px 80px;
    }

    .player-join-form {
      grid-template-rows: 90px 160px 1fr;
    }

    input {
      font-size: 4.4rem;
      font-weight: 500;
      border-bottom: 4px solid var(--primary-text-color);
      padding-top: 24px;
      padding-left: 0;
    }

    input::placeholder {
      font-weight: 500;
      font-size: 4.4rem;
    }

    .player-join-name-input {
      margin-right: 32px;
    }

    .player-join-code-input {
      margin-left: 32px;
    }
  }

  // small mobile landscape orientation
  @media screen and (max-height: 320px) {
    .player-join-form-container {
      margin-top: -20px;
    }

    .player-join-form {
      /* grid-template-rows: 60px 60px 1fr; */
      grid-template-rows: 50px 40px 1fr;
    }

    input {
      padding-top: 30px;
    }
  }

  // medium mobile landscape orientation
  @media screen and (min-height: 321px) and (max-height: 375px) {
    grid-template-rows: 60px 1fr auto;

    .player-join-form-container {
      margin-top: -20px;
    }
  }

  // large mobile landscape orientation
  @media screen and (min-height: 376px) and (max-height: 414px) {
    .player-join-form-container {
      margin-top: -10px;
    }
  }

  //extra large mobile landscape orientation
  @media screen and (min-height: 415px) and (max-height: 567px) {
    .player-join-form-container {
      margin-top: 10px;
    }

    .player-join-form {
      grid-template-rows: 60px 60px 1fr;
    }

    input {
      padding-top: 40px;
    }
  }

  // all mobile size landscape orientation
  @media screen and (max-height: 567px) {
    // header, main, footer
    grid-template-rows: 60px 1fr auto;

    .player-join-form {
      grid-template-columns: 52% 48%;
    }

    .player-join-name-input {
      grid-row: 1;
      grid-column: 1;
    }

    .player-join-code-input {
      grid-row: 1;
      grid-column: 2;
    }
  }
`;

const PlayerJoinHeader = styled(Header)`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  // CoC logo
  h1 {
    font-size: 3rem;
    line-height: 3.5rem;
    text-transform: uppercase;

    margin-bottom: -12px;
    vertical-align: bottom;
  }

  @media screen and(min-width: 786px) {
    h1 {
      font-size: 4rem;
      line-height: 4.5rem;

      margin-bottom: -15px;
    }
  }

  @media screen and (min-width: 970px) {
    h1 {
      line-height: 5rem;

      margin-bottom: -24px;
    }
  }

  @media screen and (min-width: 2560px) {
    h1 {
      font-size: 6rem;
      line-height: 6rem;

      margin-bottom: -16px;
    }
  }
`;

const PlayerJoinButton = styled(Button)`
  grid-row: 4;
  margin: 0 auto;

  width: 260px;
  height: 72px;

  border-radius: 2px;
  background-color: var(--secondary-background-color);
  font-size: 1.2rem;
  text-transform: uppercase;

  @media screen and (min-width: 646px) {
    grid-column: 1 / 3;
  }

  @media screen and (min-width: 786px) {
    grid-row: 4;
  }

  @media screen and (min-width: 970px) {
    font-size: 1.2rem;
    width: 288px;
    height: 80px;
  }

  @media screen and (min-width: 2000px) {
    font-size: 1.5rem;
    width: 360px;
    height: 100px;
  }

  @media screen and (min-width: 2560px) {
    font-size: 4rem;
    width: 900px;
    height: 225px;
  }

  // mobile landscape orientation
  @media screen and (max-height: 567px) {
    grid-column: 1 / 3;
    grid-row: 3;
  }
`;

export default function PlayerJoinScreen() {
  const { dispatch } = useContext(PlayerContext);

  const [name, setName] = useState('');
  const [joinCode, setJoinCode] = useState('');

  function handleSubmit(e) {
    e.preventDefault();
    dispatch({
      type: 'JOIN_LOBBY',
      payload: { lobbyId: joinCode, playerName: name },
    });
  }

  function handleNameChange(e) {
    const inputValue = e.target.value;

    // Filter user input to ignore special characters
    if (/\W+/g.test(inputValue)) return;

    // Only accepts alphanumeric and underscore characters
    setName(inputValue);
  }

  function handleJoinCodeChange(e) {
    const inputValue = e.target.value;

    // Filter user input to ignore numeric and special characters
    if (/[^A-Za-z]+/g.test(inputValue)) return;

    // Accepts uppercase and lowercase letter characters, but converts it all to uppercase
    setJoinCode(inputValue.toUpperCase());
  }

  return (
    <PlayerJoinContainer
      data-testid="player-join-container"
      className="primary-background"
    >
      <PlayerJoinHeader>
        <h1>Cards of Carousal</h1>
      </PlayerJoinHeader>
      <main>
        <div className="player-join-form-container">
          <form onSubmit={(e) => handleSubmit(e)} className="player-join-form">
            <input
              required
              type="text"
              maxLength="12"
              placeholder="name"
              value={name}
              onChange={handleNameChange}
              className="player-join-name-input"
            />
            <input
              required
              maxLength="4"
              type="text"
              placeholder="join code"
              value={joinCode}
              onChange={handleJoinCodeChange}
              className="player-join-code-input"
            />
            <PlayerJoinButton
              data-testid="player-join-submit-button"
              type="submit"
            >
              Let&apos;s Carouse!
            </PlayerJoinButton>
          </form>
        </div>
      </main>
      <Footer />
    </PlayerJoinContainer>
  );
}

PlayerJoinScreen.propTypes = propTypes;
