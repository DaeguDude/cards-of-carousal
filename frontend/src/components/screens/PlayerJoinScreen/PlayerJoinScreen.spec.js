import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PlayerJoinScreen from './PlayerJoinScreen';

describe('PlayerJoin', () => {
  describe('render', () => {
    it('should render', () => {
      render(<PlayerJoinScreen />);

      expect(screen.getByTestId('player-join-container')).toBeInTheDocument();
    });
  });

  describe('snapshot', () => {
    it('matches', () => {
      const tree = renderer.create(<PlayerJoinScreen />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
