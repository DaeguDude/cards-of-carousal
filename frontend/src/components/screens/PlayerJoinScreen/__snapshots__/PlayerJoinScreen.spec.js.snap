// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerJoin snapshot matches 1`] = `
.c3 {
  display: block;
  border: 2px solid var(--primary-text-color);
  font-size: 12px;
  font-weight: bold;
  padding: 0.7em;
  background: var(--secondary-background-color);
  color: var(--secondary-text-color);
}

.c3:hover,
.c3:active,
.c3:focus {
  background: var(--accent-background-color);
  cursor: pointer;
}

.c1 {
  width: 100%;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 200px;
  font-size: 3.5rem;
  font-weight: 400;
  padding-right: 16px;
  padding-bottom: 10px;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
}

.c5 p {
  text-align: right;
  position: relative;
}

.c5 a {
  color: var(--secondary-text-color);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: grid;
  height: 100%;
  background-color: var(--primary-background-color);
  grid-template-rows: 190px 1fr 140px;
  grid-template-columns: 1fr;
}

.c0 main {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .player-join-form-container {
  width: 100%;
  max-width: 1650px;
  margin: 0 auto;
  text-align: center;
  padding: 20px 80px 20px 80px;
}

.c0 .player-join-form {
  display: grid;
  grid-template-columns: 52% 48%;
  grid-template-rows: 90px 160px 1fr;
}

.c0 input {
  text-transform: uppercase;
  font-size: 4.4rem;
  font-weight: 500;
  border: none;
  border-bottom: 4px solid var(--primary-text-color);
  padding-top: 24px;
  background: transparent;
  padding-left: 0;
}

.c0 input::-webkit-input-placeholder {
  color: var(--primary-text-color);
  font-weight: 500;
  font-size: 4.4rem;
}

.c0 input::-moz-placeholder {
  color: var(--primary-text-color);
  font-weight: 500;
  font-size: 4.4rem;
}

.c0 input:-ms-input-placeholder {
  color: var(--primary-text-color);
  font-weight: 500;
  font-size: 4.4rem;
}

.c0 input::placeholder {
  color: var(--primary-text-color);
  font-weight: 500;
  font-size: 4.4rem;
}

.c0 .player-join-name-input {
  margin-right: 32px;
}

.c0 .player-join-code-input {
  margin-left: 32px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 h1 {
  font-size: 6rem;
  line-height: 6rem;
  text-transform: uppercase;
  margin-bottom: -16px;
  vertical-align: bottom;
}

.c4 {
  grid-column: 1 / 3;
  grid-row: 3;
  width: 900px;
  height: 250px;
  margin: 0 auto;
  border-radius: 2px;
  background-color: var(--secondary-background-color);
  text-transform: uppercase;
  font-size: 4rem;
}

@media (max-width:2000px) {
  .c5 {
    height: 140px;
    font-size: 2rem;
  }
}

@media (max-width:970px) {
  .c5 {
    text-align: center;
    font-size: 1.5rem;
  }
}

@media (max-width:785px) {
  .c5 {
    height: 120px;
    background-color: transparent;
  }

  .c5 p {
    text-align: center;
    font-size: 1.2rem;
    padding-bottom: 32px;
    color: var(--accent-text-color);
  }

  .c5 a {
    color: var(--accent-text-color);
    display: block;
  }
}

@media (max-width:320px) {
  .c5 {
    font-size: 1rem;
    padding-right: 0px;
  }
}

@media screen and (max-width:2000px) {
  .c0 .player-join-form-container {
    max-width: 1000px;
  }

  .c0 .player-join-form {
    grid-template-rows: 60px 88px 1fr;
  }

  .c0 input {
    font-size: 1.6rem;
    border-bottom: 2px solid var(--primary-text-color);
    padding-top: 36px;
  }

  .c0 input::-webkit-input-placeholder {
    font-size: 1.6rem;
  }

  .c0 input::-moz-placeholder {
    font-size: 1.6rem;
  }

  .c0 input:-ms-input-placeholder {
    font-size: 1.6rem;
  }

  .c0 input::placeholder {
    font-size: 1.6rem;
  }

  .c0 .player-join-name-input {
    margin-right: 16px;
  }

  .c0 .player-join-code-input {
    margin-left: 16px;
  }
}

@media screen and (max-width:1024px) {
  .c0 .player-join-form-container {
    max-width: 800px;
  }

  .c0 .player-join-form {
    grid-template-rows: 60px 80px 1fr;
  }
}

@media screen and (max-width:970px) {
  .c0 .player-join-form {
    grid-template-rows: 60px 60px 1fr;
  }

  .c0 input {
    font-size: 1.2rem;
    border-bottom: 2px solid var(--primary-text-color);
    padding-top: 40px;
  }

  .c0 input::-webkit-input-placeholder {
    font-size: 1.2rem;
  }

  .c0 input::-moz-placeholder {
    font-size: 1.2rem;
  }

  .c0 input:-ms-input-placeholder {
    font-size: 1.2rem;
  }

  .c0 input::placeholder {
    font-size: 1.2rem;
  }
}

@media screen and (max-width:785px) {
  .c0 {
    grid-template-rows: 180px 1fr 120px;
  }

  .c0 .player-join-form-container {
    margin-top: -88px;
  }
}

@media screen and (max-width:646px) {
  .c0 .player-join-form-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  .c0 .player-join-form {
    grid-template-columns: 1fr;
    grid-template-rows: 60px 60px 40px 1fr;
  }

  .c0 .player-join-name-input {
    grid-row: 1;
    grid-column: 1;
  }

  .c0 .player-join-code-input {
    grid-row: 2;
    grid-column: 1;
  }

  .c0 .player-join-name-input,
  .c0 .player-join-code-input {
    margin: 0 16px;
  }
}

@media screen and (max-width:375px) {
  .c0 .player-join-form-container {
    padding: 20px;
  }
}

@media screen and (max-width:365px) {
  .c0 {
    grid-template-rows: 130px 1fr 140px;
  }
}

@media screen and (max-width:970px) {
  .c2 h1 {
    font-size: 4rem;
    line-height: 5rem;
    margin-bottom: -20px;
  }
}

@media screen and (max-width:785px) {
  .c2 h1 {
    font-size: 4rem;
    line-height: 4.5rem;
    margin-bottom: -15px;
  }
}

@media screen and (max-width:365px) {
  .c2 h1 {
    font-size: 3rem;
    line-height: 3.5rem;
    margin-bottom: -12px;
  }
}

@media screen and (max-width:2000px) {
  .c4 {
    font-size: 1.5rem;
    width: 360px;
    height: 100px;
  }
}

@media screen and (max-width:970px) {
  .c4 {
    font-size: 1.2rem;
    width: 288px;
    height: 80px;
  }
}

@media screen and (max-width:785px) {
  .c4 {
    grid-row: 4;
  }
}

@media screen and (max-width:450px) {
  .c4 {
    width: 260px;
    height: 72px;
  }
}

<div
  className="c0 primary-background"
  data-testid="player-join-container"
>
  <header
    className="c1 c2"
  >
    <h1>
      Cards of Carousal
    </h1>
  </header>
  <main>
    <div
      className="player-join-form-container"
    >
      <form
        className="player-join-form"
        onSubmit={[Function]}
      >
        <input
          className="player-join-name-input"
          maxLength="12"
          onChange={[Function]}
          placeholder="name"
          required={true}
          type="text"
          value=""
        />
        <input
          className="player-join-code-input"
          maxLength="4"
          onChange={[Function]}
          placeholder="join code"
          required={true}
          type="text"
          value=""
        />
        <button
          className="c3 c4"
          data-testid="player-join-submit-button"
          type="submit"
        >
          Let's Carouse!
        </button>
      </form>
    </div>
  </main>
  <footer
    className="c5"
  >
    <p>
      Card content thanks to:
       
      <a
        href="https://cardsagainsthumanity.com"
      >
        https://cardsagainsthumanity.com
      </a>
    </p>
    <p>
      Made by Odin students, with love:
       
      <a
        href="https://www.theodinproject.com"
      >
        https://www.theodinproject.com
      </a>
    </p>
  </footer>
</div>
`;
